service/gcs/bridge/bridge.go

```diff
@@ -310,20 +262,22 @@ func (b *Bridge) ListenAndServe(bridgeIn io.ReadCloser, bridgeOut io.WriteCloser
 	go func() {
 		for req := range requestChan {
 			go func(r *Request) {
-				wr := &requestResponseWriter{
+				br := bridgeResponse{
 					header: &prot.MessageHeader{
 						Type: prot.GetResponseIdentifier(r.Header.Type),
 						ID:   r.Header.ID,
 					},
-					respChan: b.responseChan,
 				}
-				b.Handler.ServeMsg(wr, r)
-				if !wr.respWritten {
-					logrus.WithFields(logrus.Fields{
-						"message-type": r.Header.Type.String(),
-						"message-id":   r.Header.ID,
-					}).Error("opengcs::bridge - request didn't write response")
+				resp, err := b.Handler.ServeMsg(r)
+				if err != nil {
+					r := &prot.MessageResponseBase{
+						ActivityID: "00000000-0000-0000-0000-000000000000",
```

Comment:
Should be copied from the request?

Reply:
I wish I could. Its part of the json message not the header so if we fail json deserialization we dont have an ActivityID to respond. I have another change where I do the base unmarshal before the function invocation so we can get this as long as we get the message deserialized

Reply:
I did this anyways by doing an early deserialization.